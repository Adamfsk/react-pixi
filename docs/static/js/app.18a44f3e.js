(window.webpackJsonp=window.webpackJsonp||[]).push([[1],{"./.docz/app/db.json":function(e){e.exports={config:{title:"React Pixi",description:"Write PIXI applications using React declarative style",themeConfig:{colors:{primary:"tomato"}},ordering:"ascending",version:"0.1.2",repository:"https://github.com/inlet/react-pixi",propsParser:!1,hashRouter:!0,dest:"docs"},entries:{"README.mdx":{name:"About React Pixi",order:0,route:"/",id:"075c8e9dd56c748f3da128a33861a770",filepath:"README.mdx",link:"https://github.com/inlet/react-pixi/edit/master/README.mdx",slug:"readme",menu:null,headings:[{depth:1,slug:"react-fiber-renderer-for-pixi",value:"React fiber renderer for pixi"},{depth:2,slug:"install",value:"Install"},{depth:3,slug:"node",value:"Node"},{depth:3,slug:"browser",value:"Browser"},{depth:2,slug:"usage",value:"Usage"},{depth:4,slug:"with-react-dom",value:"With react dom"},{depth:4,slug:"without-react-dom",value:"Without react dom"},{depth:2,slug:"examples",value:"Examples"},{depth:2,slug:"custom-components",value:"Custom components"},{depth:2,slug:"access-pixi-application",value:"Access pixi application"},{depth:3,slug:"via-render-props-function-as-child",value:"Via render props function as child"},{depth:3,slug:"via-higher-order-component-withpixiapp",value:"Via higher order component withpixiapp"},{depth:2,slug:"stage",value:"Stage"},{depth:3,slug:"props",value:"Props"},{depth:2,slug:"components",value:"Components"}]},"src/components/BitmapText.mdx":{name:"BitmapText",order:10,menu:"Components",route:"/components/bitmap-text",id:"be3209d167db8b7dbffcef178bad87e7",filepath:"src/components/BitmapText.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/BitmapText.mdx",slug:"src-components-bitmap-text",headings:[{depth:1,slug:"bitmaptext",value:"Bitmaptext"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Container.mdx":{name:"Container",menu:"Components",route:"/components/container",order:1,id:"2e5bc2a2cd68ab9f0f8d3df1f4d34462",filepath:"src/components/Container.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Container.mdx",slug:"src-components-container",headings:[{depth:1,slug:"container",value:"Container"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Graphics.mdx":{name:"Graphics",order:5,menu:"Components",route:"/components/graphics",id:"46f51752305a331c95688ef0dfbac9f1",filepath:"src/components/Graphics.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Graphics.mdx",slug:"src-components-graphics",headings:[{depth:1,slug:"graphics",value:"Graphics"},{depth:2,slug:"props",value:"Props"},{depth:3,slug:"draw",value:"Draw"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Mesh.mdx":{name:"Mesh",order:6,menu:"Components",route:"/components/mesh",id:"241af672f33b0e74e7740e76058da4b7",filepath:"src/components/Mesh.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Mesh.mdx",slug:"src-components-mesh",headings:[{depth:1,slug:"mesh",value:"Mesh"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/NineSlicePlane.mdx":{name:"NineSlicePlane",order:8,menu:"Components",route:"/components/mesh/nine-slice-plane",id:"6af4bc991617b1261af6cd7a06121fcf",filepath:"src/components/NineSlicePlane.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/NineSlicePlane.mdx",slug:"src-components-nine-slice-plane",headings:[{depth:1,slug:"ninesliceplane",value:"Ninesliceplane"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/ParticleContainer.mdx":{name:"ParticleContainer",order:2,menu:"Components",route:"/components/particle-container",id:"b241ab7afa23042ecb1ca32bf94c7476",filepath:"src/components/ParticleContainer.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/ParticleContainer.mdx",slug:"src-components-particle-container",headings:[{depth:1,slug:"particlecontainer",value:"Particlecontainer"},{depth:2,slug:"props",value:"Props"},{depth:3,slug:"maxsize",value:"Maxsize"},{depth:3,slug:"properties",value:"Properties"},{depth:3,slug:"batchsize",value:"Batchsize"},{depth:3,slug:"autoresize",value:"Autoresize"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Rope.mdx":{name:"Rope",order:7,menu:"Components",route:"/components/mesh/rope",id:"a42b1d058c5c14224da80287ee2a57cc",filepath:"src/components/Rope.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Rope.mdx",slug:"src-components-rope",headings:[{depth:1,slug:"rope",value:"Rope"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Sprite.mdx":{name:"Sprite",order:3,menu:"Components",route:"/components/sprite",id:"a85d3bf3e3eb2bc95c10eef22bbe23cf",filepath:"src/components/Sprite.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Sprite.mdx",slug:"src-components-sprite",headings:[{depth:1,slug:"sprite",value:"Sprite"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Text.mdx":{name:"Text",order:9,menu:"Components",route:"/components/text",id:"45ae8ab60e42d20e40bf547ae288fbc2",filepath:"src/components/Text.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Text.mdx",slug:"src-components-text",headings:[{depth:1,slug:"text",value:"Text"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"style-props",value:"Style props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/TilingSprite.mdx":{name:"TilingSprite",order:4,menu:"Components",route:"/components/tiling-sprite",id:"96901b8d151744c970684916804051db",filepath:"src/components/TilingSprite.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/TilingSprite.mdx",slug:"src-components-tiling-sprite",headings:[{depth:1,slug:"tilingsprite",value:"Tilingsprite"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/stage/Stage.mdx":{name:"Stage",order:1,route:"/stage",id:"d912e9fd43740cdfc8c6e92ca9e55715",filepath:"src/stage/Stage.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/stage/Stage.mdx",slug:"src-stage-stage",menu:null,headings:[{depth:1,slug:"stage",value:"Stage"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]}}}},"./.docz/app/index.jsx":function(e,t,n){"use strict";n.r(t);var s=n("./node_modules/react/index.js"),o=n.n(s),a=n("./node_modules/react-dom/index.js"),p=n.n(a),i={"README.mdx":function(){return n.e(2).then(n.bind(null,"./README.mdx"))},"src/components/BitmapText.mdx":function(){return Promise.all([n.e(0),n.e(3)]).then(n.bind(null,"./src/components/BitmapText.mdx"))},"src/components/Container.mdx":function(){return Promise.all([n.e(0),n.e(4)]).then(n.bind(null,"./src/components/Container.mdx"))},"src/components/Graphics.mdx":function(){return Promise.all([n.e(0),n.e(5)]).then(n.bind(null,"./src/components/Graphics.mdx"))},"src/components/Mesh.mdx":function(){return Promise.all([n.e(0),n.e(6)]).then(n.bind(null,"./src/components/Mesh.mdx"))},"src/components/NineSlicePlane.mdx":function(){return Promise.all([n.e(0),n.e(7)]).then(n.bind(null,"./src/components/NineSlicePlane.mdx"))},"src/components/ParticleContainer.mdx":function(){return Promise.all([n.e(0),n.e(8)]).then(n.bind(null,"./src/components/ParticleContainer.mdx"))},"src/components/Rope.mdx":function(){return Promise.all([n.e(0),n.e(9)]).then(n.bind(null,"./src/components/Rope.mdx"))},"src/components/Sprite.mdx":function(){return Promise.all([n.e(0),n.e(10)]).then(n.bind(null,"./src/components/Sprite.mdx"))},"src/components/Text.mdx":function(){return Promise.all([n.e(0),n.e(11)]).then(n.bind(null,"./src/components/Text.mdx"))},"src/components/TilingSprite.mdx":function(){return Promise.all([n.e(0),n.e(12)]).then(n.bind(null,"./src/components/TilingSprite.mdx"))},"src/stage/Stage.mdx":function(){return Promise.all([n.e(0),n.e(13)]).then(n.bind(null,"./src/stage/Stage.mdx"))}},r=n("./.docz/app/root.jsx"),l=[],c=[],d=function(){return c.forEach(function(e){return e&&e()})},m=document.querySelector("#root");(function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:r.a;l.forEach(function(e){return e&&e()}),p.a.render(o.a.createElement(e,{imports:i}),m,d)})(r.a)},"./.docz/app/root.jsx":function(e,t,n){"use strict";(function(e){var s=n("./node_modules/react/index.js"),o=n.n(s),a=n("./node_modules/react-hot-loader/index.js"),p=n("./node_modules/docz-theme-default/dist/index.m.js"),i=n("./.docz/app/db.json");t.a=Object(a.hot)(e)(function(e){var t=e.imports;return o.a.createElement(p.a,{db:i,imports:t,hashRouter:!0})})}).call(this,n("./node_modules/webpack/buildin/harmony-module.js")(e))},0:function(e,t,n){n("./node_modules/docz-core/node_modules/babel-polyfill/lib/index.js"),e.exports=n("./.docz/app/index.jsx")}},[[0,14,0]],[0,3,4,5,6,7,8,9,10,11,12,13,2]]);