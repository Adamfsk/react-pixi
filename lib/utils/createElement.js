import { BitmapText, Container, Graphics, NineSlicePlane, ParticleContainer, Sprite, Stage, Text, TilingSprite, Mesh, Rope } from '../components';
export var TYPES = {
  BitmapText: 'BitmapText',
  Container: 'Container',
  Graphics: 'Graphics',
  NineSlicePlane: 'NineSlicePlane',
  ParticleContainer: 'ParticleContainer',
  Sprite: 'Sprite',
  Stage: 'Stage',
  Text: 'Text',
  TilingSprite: 'TilingSprite',
  Shape: 'Shape',
  Mesh: 'Mesh',
  Rope: 'Rope'
};
export function createElement(type, props, root) {
  var ELEMENTS = {
    BitmapText: BitmapText,
    Container: Container,
    Graphics: Graphics,
    NineSlicePlane: NineSlicePlane,
    ParticleContainer: ParticleContainer,
    Sprite: Sprite,
    Stage: Stage,
    Mesh: Mesh,
    Rope: Rope,
    Text: Text,
    TilingSprite: TilingSprite,
    default: undefined
  };
  var ins = ELEMENTS[type];
  return ins && typeof ins === 'function' ? ins(root, props || {}) : ELEMENTS.default;
}