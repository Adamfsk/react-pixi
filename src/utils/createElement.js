import {
  BitmapText,
  Container,
  Graphics,
  NineSlicePlane,
  ParticleContainer,
  Sprite,
  Stage,
  Text,
  TilingSprite,
} from '../components'

export const TYPES = {
  BitmapText: 'BitmapText',
  Container: 'Container',
  Graphics: 'Graphics',
  NineSlicePlane: 'NineSlicePlane',
  ParticleContainer: 'ParticleContainer',
  Sprite: 'Sprite',
  Stage: 'Stage',
  Text: 'Text',
  TilingSprite: 'TilingSprite',
  Shape: 'Shape',
  Mesh: 'Mesh',
  Rope: 'Rope',
}

export function createElement(type, props, root) {
  const ELEMENTS = {
    BitmapText,
    Container,
    Graphics,
    NineSlicePlane,
    ParticleContainer,
    Sprite,
    Stage,
    Text,
    TilingSprite,
    default: undefined,
  }

  const ins = ELEMENTS[type]
  return ins && typeof ins === 'function' ? ins(root, props) : ELEMENTS.default
}
