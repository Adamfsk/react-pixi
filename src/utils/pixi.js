import * as PIXI from 'pixi.js'
import idx from 'idx'
import isNil from 'lodash/isNil'

/**
 * Parse PIXI point value to array of coordinates
 *
 * @param {*} value
 * @returns {Array}
 */
export function parsePoint(value) {
  let arr = []

  if (typeof value === 'undefined') {
    return arr
  } else if (typeof value === 'string') {
    arr = value.split(',')
  } else if (typeof value === 'number') {
    arr = [value]
  } else if (Array.isArray(value)) {
    arr = [...value]
  } else if (value !== null && typeof value === 'object') {
    const x = idx(value, _ => _.x) || 0
    const y = idx(value, _ => _.y) || 0
    arr = [x, y]
  } else {
    return arr
  }

  return arr.filter(p => !isNil(p) && !isNaN(p)).map(Number)
}

/**
 * Determine value is type of Point or ObservablePoint
 *
 * @param {*} value
 * @returns {boolean}
 */
export function isPointType(value) {
  return value instanceof PIXI.Point || value instanceof PIXI.ObservablePoint
}

/**
 * Event handlers
 *
 * @type {string[]}
 */
export const eventHandlers = [
  'click',
  'mousedown',
  'mousemove',
  'mouseout',
  'mouseover',
  'mouseup',
  'mouseupoutside',
  'tap',
  'touchstart',
  'touchmove',
  'touchend',
  'touchendoutside',
  'pointercancel',
  'pointerout',
  'pointerover',
  'pointertap',
  'pointerdown',
  'pointerup',
  'pointerupoutside',
  'pointermove',
  'rightclick',
  'rightdown',
  'rightup',
  'rightupoutside',
  'touchcancel',
  'touchendoutside',
  'touchmove',
  'touchstart',
]
