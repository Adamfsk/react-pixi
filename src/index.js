import { TYPES } from './utils/element'
import { render } from './render'
import Stage from './stage'
import Provider from './stage/provider'

/**
 * -------------------------------------------
 * Public API
 * -------------------------------------------
 */

export { render, Stage, Provider }

export const BitmapText = TYPES.BitmapText
export const Container = TYPES.Container
export const Graphics = TYPES.Graphics
export const NineSlicePlane = TYPES.NineSlicePlane
export const ParticleContainer = TYPES.ParticleContainer
export const Sprite = TYPES.Sprite
export const Text = TYPES.Text
export const TilingSprite = TYPES.TilingSprite
export const Shape = TYPES.Shape
export const Mesh = TYPES.Mesh
export const Rope = TYPES.Rope
