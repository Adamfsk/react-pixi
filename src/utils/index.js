import { createElement } from './createElement'
import getTextureFromProps from './getTextureFromProps'
import defaults from './defaults'

export { createElement, getTextureFromProps, defaults }
