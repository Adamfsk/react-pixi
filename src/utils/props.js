import * as PIXI from 'pixi.js'
import invariant from 'fbjs/lib/invariant'

/**
 * Reserved props
 *
 * @type {Object}
 */
export const PROPS_RESERVED = {
  children: true,
  parent: true,
  worldAlpha: true,
  worldTransform: true,
  worldVisible: true,
}

/**
 * Default display object props
 *
 * @type {Object}
 */
export const PROPS_DISPLAY_OBJECT = {
  alpha: 1,
  buttonMode: false,
  cacheAsBitmap: false,
  cursor: 'auto',
  filterArea: null,
  filters: null,
  hitArea: null,
  interactive: false,
  mask: null,
  pivot: 0,
  position: 0,
  renderable: true,
  rotation: 0,
  scale: 1,
  skew: 0,
  transform: null,
  visible: true,
  x: 0,
  y: 0,
}

/**
 * Helper util for fetching the texture from props
 * Can be either texture or image
 *
 * @param {string} elementType
 * @param {PIXI.Texture} texture
 * @param {string|undefined} image
 * @returns {PIXI.Texture}
 */
export const getTextureFromProps = (elementType, { texture = undefined, image = undefined }) => {
  if (image) {
    invariant(typeof image === 'string', elementType + ' image needs to be a string, got `%s`', typeof image)
    return PIXI.Texture.fromImage(image)
  }

  invariant(texture instanceof PIXI.Texture, elementType + ' texture needs to be type of `PIXI.Texture`')
  return texture
}
