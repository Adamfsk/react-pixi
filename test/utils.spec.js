import React from 'react'
import * as PIXI from 'pixi.js'
import { createElement, TYPES } from '../src/utils/createElement'
import { props, pixi } from '../src/utils'
import { getTextureFromProps } from '../src/utils/props'

import { emptyTexture } from './__fixtures__/textures'
import { desyrel } from './__fixtures__/bitmapfonts'
import parseBitmapFont from './__utils__/parseBitmapFont'

const noop = () => {}

parseBitmapFont(desyrel)

describe('props', () => {
  test('reserved props', () => {
    expect(props.PROPS_RESERVED).toMatchSnapshot()
  })
  test('display object props', () => {
    expect(props.PROPS_DISPLAY_OBJECT).toMatchSnapshot()
  })
})

describe('createElement', () => {
  test('types', () => {
    expect(TYPES).toMatchSnapshot()
  })

  test('create Container', () => {
    const element = createElement(TYPES.Container)
    expect(element).toBeInstanceOf(PIXI.Container)
  })

  test('create Text', () => {
    const element = createElement(TYPES.Text, { text: 'foobar' })
    expect(element).toBeInstanceOf(PIXI.Text)
  })

  test('create Sprite', () => {
    const element = createElement(TYPES.Sprite, { texture: emptyTexture })
    expect(element).toBeInstanceOf(PIXI.Sprite)
  })

  test('create ParticleContainer', () => {
    const element = createElement(TYPES.ParticleContainer)
    expect(element).toBeInstanceOf(PIXI.particles.ParticleContainer)
  })

  test('create BitmapText', () => {
    const element = createElement(TYPES.BitmapText, { text: 'foobar', style: { font: '35px Desyrel' } })
    expect(element).toBeInstanceOf(PIXI.extras.BitmapText)
  })

  test('create TilingSprite', () => {
    const element = createElement(TYPES.TilingSprite, { texture: emptyTexture })
    expect(element).toBeInstanceOf(PIXI.extras.TilingSprite)
  })

  test('create Graphics', () => {
    const element = createElement(TYPES.Graphics)
    expect(element).toBeInstanceOf(PIXI.Graphics)
  })

  test('create NineSlicePlane', () => {
    const element = createElement(TYPES.NineSlicePlane, { texture: emptyTexture })
    expect(element).toBeInstanceOf(PIXI.mesh.NineSlicePlane)
  })

  test('create Mesh', () => {
    const element = createElement(TYPES.Mesh, { texture: emptyTexture })
    expect(element).toBeInstanceOf(PIXI.mesh.Mesh)
  })

  test('create Rope', () => {
    const element = createElement(TYPES.Rope, { texture: emptyTexture, points: [] })
    expect(element).toBeInstanceOf(PIXI.mesh.Rope)
  })

  test('get undefined', () => {
    expect(createElement('INVALID')).toBeUndefined()
  })
})

describe('getTextureFromProps', () => {
  let spy

  beforeAll(() => {
    spy = jest.spyOn(PIXI.Texture, 'fromImage').mockReturnValue(emptyTexture)
  })

  afterAll(() => {
    spy.mockRestore()
  })

  test('invariant image', () => {
    expect(() => getTextureFromProps('Test', { image: 123 })).toThrow('Test image needs to be a string, got `number`')
  })

  test('invariant texture', () => {
    expect(() => getTextureFromProps('Test', { texture: 'texture' })).toThrow(
      'Test texture needs to be type of `PIXI.Texture`'
    )
  })

  test('get texture from image', () => {
    const texture = getTextureFromProps('Test', { image: './image.png' })
    expect(texture).toBeInstanceOf(PIXI.Texture)
    expect(spy).toBeCalledWith('./image.png')
  })

  test('get texture from texture', () => {
    const texture = getTextureFromProps('Test', { texture: emptyTexture })
    expect(texture).toBe(emptyTexture)
  })
})

describe('pixi', () => {
  describe('parsePoint', () => {
    test('parse undefined', () => {
      expect(pixi.parsePoint(undefined)).toEqual([])
    })

    test('parse null', () => {
      expect(pixi.parsePoint(null)).toEqual([])
    })

    test('parse string', () => {
      expect(pixi.parsePoint('1,3')).toEqual([1, 3])
    })

    test('parse invalid string', () => {
      expect(pixi.parsePoint('not, valid')).toEqual([])
    })

    test('parse number', () => {
      expect(pixi.parsePoint(100)).toEqual([100])
    })

    test('parse shallow array', () => {
      expect(pixi.parsePoint([100, 200])).toEqual([100, 200])
      expect(pixi.parsePoint([100, 200])).not.toBe([100, 200])
    })

    test('parse object with x y', () => {
      expect(pixi.parsePoint({ x: 100, y: 200 })).toEqual([100, 200])
    })

    test('parse object with x only', () => {
      expect(pixi.parsePoint({ x: 100 })).toEqual([100, 0])
    })

    test('parse object with y only', () => {
      expect(pixi.parsePoint({ y: 200 })).toEqual([0, 200])
    })
  })

  describe('isPointType', () => {
    describe('true', () => {
      test('point', () => {
        expect(pixi.isPointType(new PIXI.Point(100, 200))).toBeTruthy()
      })

      test('observablepoint', () => {
        expect(pixi.isPointType(new PIXI.ObservablePoint(noop, this, 100, 200))).toBeTruthy()
      })
    })

    describe('false', () => {
      test('string', () => {
        expect(pixi.isPointType(undefined)).toBeFalsy()
      })

      test('number', () => {
        expect(pixi.isPointType(123)).toBeFalsy()
      })

      test('array', () => {
        expect(pixi.isPointType([100, 200])).toBeFalsy()
      })

      test('object', () => {
        expect(pixi.isPointType({ x: 100, y: 200 })).toBeFalsy()
      })
    })
  })

  describe('eventHandlers', () => {
    test('available event hanlders', () => {
      expect(pixi.eventHandlers).toMatchSnapshot()
    })
  })
})
