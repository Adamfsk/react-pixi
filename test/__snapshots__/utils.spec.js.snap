// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createElement types 1`] = `
Object {
  "BitmapText": "BitmapText",
  "Container": "Container",
  "Graphics": "Graphics",
  "Mesh": "Mesh",
  "NineSlicePlane": "NineSlicePlane",
  "ParticleContainer": "ParticleContainer",
  "Rope": "Rope",
  "Shape": "Shape",
  "Sprite": "Sprite",
  "Stage": "Stage",
  "Text": "Text",
  "TilingSprite": "TilingSprite",
}
`;

exports[`defaults display object props 1`] = `
Object {
  "alpha": 1,
  "buttonMode": false,
  "cacheAsBitmap": false,
  "cursor": "auto",
  "filterArea": null,
  "filters": null,
  "hitArea": null,
  "interactive": false,
  "mask": null,
  "pivot": 0,
  "position": 0,
  "renderable": true,
  "rotation": 0,
  "scale": 1,
  "skew": 0,
  "transform": null,
  "visible": true,
  "x": 0,
  "y": 0,
}
`;

exports[`defaults reserved props 1`] = `
Object {
  "children": true,
  "parent": true,
  "worldAlpha": true,
  "worldTransform": true,
  "worldVisible": true,
}
`;
